src/hooks/relay/useSelectedRelayGroup.ts

```diff
@@ -0,0 +1,58 @@
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { useReadonlyMyPublicKey } from 'hooks/useMyPublicKey';
+import { useRelayGroupsQuery } from './useRelayGroupsQuery';
+import { useCallback } from 'react';
+
+const SELECTED_RELAY_GROUP_ID_KEY =
+  'relay-selector:selected-group-id:{{pubKey}}';
+
+const getLegacyLocalValue = (key: string) => {
+  if (typeof window === 'undefined') {
+    return;
+  }
+
+  const jsonVal = localStorage.getItem(key);
+  if (jsonVal) {
+    try {
+      const value = JSON.parse(jsonVal);
+      return value;
+    } catch (e) {
+      localStorage.setItem(key, JSON.stringify(jsonVal));
+      return jsonVal;
+    }
+  }
+};
+
+export function useSelectedRelayGroup() {
+  const myPublicKey = useReadonlyMyPublicKey();
```

Comment:
do you remember we have a selected relay local storing issue before? I impl a fix at that time https://github.com/digi-monkey/flycat-web/commit/e0f3cd60a4a3b80ed0645122cbada1466db73dbd# should we use redux publickey here to prevent the same issue?

Reply:
I think “use redux publicly” should be built-in within the “useReadonlyMyPublicKey” hook, instead of duplicating the same logic everywhere.

Reply:
Or maybe impl new hook `useMyPublicKey` to prevent this issue

Reply:
yes  I agree, `useReadonlyMyPublicKey` is just some legacy code needs to be updated

Reply:
I think new hook `useMyPublicKey` is good. we can now only replace here with `useMyPublicKey`, and then to replace all the `useReadonlyMyPublicKey` with this new hooks later, if we got more time.

Reply:
Consider handling the public key in another PR.
For now, use the redux public key to fix it.